Lombiq.HelpfulLibraries/Libraries/Utilities/MulticastDelegateExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace System
+{
+    public static class MulticastDelegateExtensions
+    {
+        /// <summary>
+        /// Invokes all async delegates in the given <see cref="MulticastDelegate"/> sequentially.
+        /// </summary>
+        /// <typeparam name="TDelegate">Type of the delegates in the given <see cref="MulticastDelegate"/> .</typeparam>
+        /// <param name="delegateExecution">The function to execute each delegate with.</param>
+        /// <returns>The <see cref="Task"/> that'll complete when all items have completed.</returns>
+        public static Task InvokeAsync<TDelegate>(
+            this MulticastDelegate multicastDelegate,
+            Func<TDelegate, Task> delegateExecution)
+            where TDelegate : Delegate =>
+            multicastDelegate == null

```

Comment:
Didn't we decide not to add null checks to this in extension methods to avoid inconsistencies?

Reply:
I was under the same impression.

Reply:
Yes but there since this is async, you need a traditional `if (a != null)` check every single time (which is five times in the UI Testing Toolbox).

Reply:
Why, does `await (... ?? Task.CompletedTask)` not work? We really need an `await?` operator...

Reply:
That works too but IMO it's worse (longer, does an unnecessary await, perhaps also a state machine if it's not optimized away; the latter is done by the current implementation as well but it's a sacrifice for the fluency of the consumer code).